src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -597,6 +594,59 @@ public void filterForCourseDashboard(List<StudentParticipation> participations,
         }
     }
 
+    /**
+     * Filter for appropriate submission. Relevance in the following order:
+     * - submission with rated result
+     * - submission with unrated result (late submission)
+     * - no submission with any result > latest submission
+     *
+     * @param submissions that need to be filtered
+     * @return filtered submission
+     */
+    private Submission findAppropriateSubmission(Set<Submission> submissions) {
+        List<Submission> submissionsWithRatedResult = new ArrayList<>();
+        List<Submission> submissionsWithUnratedResult = new ArrayList<>();
+        List<Submission> submissionsWithoutResult = new ArrayList<>();
+
+        if (submissions != null) {
```

Comment:
If submissions are null we can immediately return null to avoid nesting

Reply:
Thanks!